src/django-app/shops/views.py

```diff
@@ -173,6 +175,31 @@ def get_serializer_class(self):
         if self.action in ["update", "partial_update"]:
             return ServiceUpdateSerializer
         return ServiceSerializer
+    
+    @transaction.atomic
+    def create(self, request, *args, **kwargs):
+        try:
+            with transaction.atomic():
+                data = request.data
+                parts = data.pop("parts", None)
+                shop = data.get("shop")
+                service_serializer = ServiceSerializer(data=data)
+                if service_serializer.is_valid(raise_exception=True):
+                    service_serializer.save()
+                    if parts is not None:
+                        service_parts_to_create = Part.objects.filter(
+                            id__in=parts
+                        )
+                        for part in service_parts_to_create:
+                            service_part_serializer = ServicePartSerializer(shop=shop, part=part.pk, price=part.price, quantity=1)
+                            if service_part_serializer.is_valid(raise_exception=True):
+                                service_part_serializer.save()
+                return Response(service_serializer.data)
+        except Exception as err:
+            return Response(
+                {"status": False, "error_description": err.message},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
```

Comment:
@tevisdoe This looks good. The only bug I see here is "err.message". I found out recently that if you catch an "Exception", it does not have a `message` attribute. However, if you catch other exceptions like ValidationError, etc., they do have a message. So for now, you can just write a random message.

Reply:
Gotcha, gotcha. Thanks for the quick reply.